frontend/src/components/createProperty/CreateLocker.js

```diff
@@ -10,79 +11,181 @@ import {
 } from "react-bootstrap";
 
 const CreateLocker = () => {
-
+  const {properties} = useProperty();
+  // receive the containing property id from the url
   let { propertyId } = useParams();
   const navigate = useNavigate();
-  // unit information, should extend to match all info needed
+  
+  // locker information should extend to match all info needed
   const [formData, setFormData] = useState({
     public_profile: null,
     location: "",
     purchase_price: "",
     rent_price: "",
     size: "",
-    extra_information: null,
+    extra_information: "",
   });
 
+  const [errors, setErrors] = useState({});
+
+  const findPropertyById = (propertyId) => {
+    if (Array.isArray(properties)) {
+      return properties.find(property => property.id.toString() === propertyId);
+    }
+  };
+  
+  //handle change from the user input and save to state
   const handleChange = (e) => {
     setFormData({
       ...formData,
       // Trimming any whitespace
       [e.target.name]: e.target.value,
     });
+
+    //Clear existing error if there is change to input
+    setErrors({ ...errors, [e.target.name]: '' });
     console.log(formData);
   };
 
-  function goBack() {
-    navigate(-1);
-  }
+  //Validate input form
+  const validateForm = () =>{
+    const errors = {};
+    let isValid = true;
+
+    //Validate input fields 
+    //Location must be filled, length less than 4 chars
+    if (!formData.location){
+      errors.location = "Location field required";
+      isValid = false;
+    } else if (formData.location.length > 4){
+      errors.location = "Location field must be under 4 characters";
+      isValid = false;
+    }
+    
+    //Purchase price must be filled, must be a non-negative number, can only have 2 decimal places, and less than 20 characters
+    if (!formData.purchase_price){
+      errors.purchase_price = "Purchase price required";
+      isValid = false;
+    } else if (isNaN(formData.purchase_price)){
+      errors.purchase_price = "Purchase price must be a number";
+      isValid = false;
+    } else if (parseFloat(formData.purchase_price) < 0){
+      errors.purchase_price = "Invalid number";
+      isValid = false;
+    } else if (!formData.purchase_price.match(/^(\d*\.{0,1}\d{0,2}$)/)){
+      errors.purchase_price = "Purchase price can only have 2 decimal places";
+      isValid = false;
+    } else if (formData.purchase_price > 20){
```

Comment:
This verification is not consistent with the error message

Reply:
Agree! a bug task was create to address this issue ( #158) ! Thanks